Commands/GenerateVisitsDb.php

```diff
@@ -0,0 +1,525 @@
+            'Conversions generated    ' . str_pad($this->formatter->format($summaryInfo['conversions']), 12, ' ', STR_PAD_LEFT),
+            'Visits per second        ' . str_pad($this->formatter->format(round($summaryInfo['visits'] / $summaryInfo['timeTaken'], 0)), 12, ' ', STR_PAD_LEFT) . " / sec",
+            'Queries per second       ' . str_pad($this->formatter->format(round(($summaryInfo['visits'] + $summaryInfo['visitActions'] + $summaryInfo['actions'] +
+                        $summaryInfo['conversions']) / $summaryInfo['timeTaken'], 0)), 12, ' ', STR_PAD_LEFT) . ' / sec'
+            ];
+
+        $this->writeSuccessMessage($summary);
+    }
+
+    /**
+     * Return the number of visits to be created for a day, includes randomization and default values if required
+     *
+     * @return int
+     */
+    private function getLimitVisits(): int
+    {
+        $input = $this->getInput();
+        if ($input->getOption('limit-visits')) {
+            $limit = $input->getOption('limit-visits');
+            $randomPercent = $input->getOption('limit-random-percent');
+            if ($randomPercent > 0) {
+                $limit = rand(floor($limit - ($limit * ($randomPercent / 100))), ceil($limit + ($limit * ($randomPercent / 100))));
+            }
+            return $limit;
+        }
+        return rand(400, 1000);
+    }
+
+    /**
+     * Generates visits and related data for a specific day
+     *
+     * @param int $time     Start time of the day for visits to be generated
+     * @param int $idSite   Site for which to generate the visits
+     * @param int $limit    Number of visits to generate
+     *
+     * @return array        Statistics array for the data that was generated
+     *                      ['visits' => 0, 'visitActions' => 0, 'actions' => 0, 'conversions' => 0]
+     * @throws \Exception
+     */
+    private function generate(int $time, int $idSite, int $limit): array
+    {
+        $input = $this->getInput();
+        $output = $this->getOutput();
+        $conversionPercent = $input->getOption('conversion-percent');
+        $actionCountMin = $input->getOption('min-actions');
+        $actionCountMax = $input->getOption('max-actions');
+
+        $requestCount = 0;
+        $queryGenerator = new VisitFakeQuery($this->actionsPoolSize);
+
+        if ($this->verbosity == 0) {
+            $output->write(".");
+        }
+
+        $stats = [
+            'visits' => 0,
+            'visitActions' => 0,
+            'actions' => 0,
+            'conversions' => 0
+            ];
+
+        while ($requestCount < $limit || $limit < 0) {
+
+            $this->query(['sql' => 'START TRANSACTION', 'bind' => []]);
+            // Choose a random timestamp within supplied date range
+            $timestampUTC = rand($time, $time + 86000); // no new visits will start for 400 seconds before midnight
```

Comment:
Would we want some visits that span over two days to be closer to real world visits or does the current tracking also record two visits, one in each day?

Reply:
Currently visits are always reset at midnight by the tracker, so even if there is an in-progress visit with actions in the last 30mins, the first action in the new day will be treated as a new visit.